### DISCUSSION TITLE
webpage remains blank for an extended period due to slow network

### DISCUSSION THREAD

**Discussion by yevhen-harmonizehr** [argoproj/argo-workflows#11970] [Q&A] [answered]
Greetings,

My team has observed that occasionally, particularly when the cache expires, the web UI page may take a considerable amount of time to load on a slow internet connection. Upon investigation, I have identified the main.<random-hash>.js file, which is 1.5MB in size, and it could take anywhere from 10 to 30 seconds to load on a slow network like 3G. During this loading time, the user experiences a blank page.

I am exploring options to enhance the user experience. One approach I've considered is setting the gzip-level to 9 to further compress this file, reducing its size. But i can not see any option to do this on argo-workflows server.
Are there any other suggestions or techniques to eliminate this waiting period and enhance the user experience?

**Comment by agilgur5** [comment, marked as answer]
Unfortunately there's nothing built-in for this right now.

You are correct that the Argo Server has [no configuration](https://argoproj.github.io/argo-workflows/cli/argo_server/) for gzip-level right now.

I do think the bundle is a bit too large, and would like to optimize that. One method I described to do so is code-splitting in #12059.

There are potentially other bundle optimizations we can make as well; I looked into it a bit in https://github.com/argoproj/argo-workflows/issues/12059#issuecomment-1773859379
